yt/utilities/parallel_tools/parallel_analysis_interface.py

```diff
@@ -326,6 +326,32 @@ def barrierize(*args, **kwargs):
     return barrierize
 
 
+def parallel_root_only_then_broadcast(func):
+    """
+    This decorator blocks and calls the funciton on the root processor and then
+    broadcasts the results to the other processors.
+    """
+
+    @wraps(func)
+    def root_only(*args, **kwargs):
+        if not parallel_capable:
+            return func(*args, **kwargs)
+        comm = _get_comm(args)
+        rv0 = None
+        if comm.rank == 0:
+            try:
+                rv0 = func(*args, **kwargs)
+            except Exception:
+                traceback.print_last()
+        rv = comm.mpi_bcast(rv0)
+        if not rv:
+            raise RuntimeError
+        rv = comm.mpi_bcast(rv0)
```

Comment:
this line looks dupplicated ?

Reply:
yes, it is! will delete. was left over from a copy/paste from an initial attempt.

Reply:
done